File: solutions/ocaml/bob/2/bob.ml
let response_for input =
  let open Base.String in
  let input = strip input in
  let question = is_suffix ~suffix:"?" input in
  let shouting = uppercase input = input && lowercase input <> input in
  let silence = is_empty input in

  if shouting && question then "Calm down, I know what I'm doing!"
  else if shouting then "Whoa, chill out!"
  else if question then "Sure."
  else if silence then "Fine. Be that way!"
  else "Whatever."

File: solutions/fsharp/bob/2/Bob.fs
module Bob

open System

let isQuestion (input: string) =
    match input |> Seq.tryLast with
    | Some '?' -> true
    | _ -> false

let isYelling (input: string) =
    let upperChars = input |> String.filter (Char.IsUpper)
    let lowerChars = input |> String.filter (Char.IsLower)

    lowerChars.Length = 0 && upperChars.Length > 0

let response (input: string) =
    match input.Trim() with
    | x when isQuestion x && isYelling x -> "Calm down, I know what I'm doing!"
    | x when isQuestion x -> "Sure."
    | x when isYelling x -> "Whoa, chill out!"
    | "" -> "Fine. Be that way!"
    | _ -> "Whatever."
